internal/security/proxy/config/config.go

```diff
@@ -87,7 +88,7 @@ type SecretServiceInfo struct {
 
 func (s SecretServiceInfo) GetSecretSvcBaseURL() string {
 	url := &url.URL{
-		Scheme: "https",
+		Scheme: s.Protocol,
```

Comment:
Came we simplify and remove using URL and just use Sprintf. URL is only provider the formatting we can do with Sprintf
```
return fmt.Sprintf("%s://%s:%d", s.Protocol, s.Server, s.Port)
```

Reply:
sounds good to me